ruby/red-arrow/lib/arrow/table.rb

```diff
@@ -426,6 +426,14 @@ def remove_column(name_or_index)
       remove_column_raw(index)
     end
 
+    # Return column names in an Array.
+    #
+    # @return [::Array<String>] column names.
+    # 
+    def column_names
+      columns.map(&:name)
```

Comment:
How about caching the result because column information is immutable.

```suggestion
      @column_names ||= columns.map(&:name)
```

FYI: I prefer `ma` to `collect`. :-)

Reply:
I prefer collect/select/reject before. I am losing map's shortness in these days. Thank you.